modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -475,37 +430,106 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
       addMeshSubgenerator("PlaneIDMeshGenerator", build_mesh_name, params);
     }
   }
-  else
-    declareMeshProperty("extruded", false);
+
+  generateMetadata();
 
   _build_mesh = &getMeshByName(build_mesh_name);
 }
 
-std::unique_ptr<MeshBase>
-PinMeshGenerator::generate()
+void
+PinMeshGenerator::generateMetadata()
 {
-  // Must be called to free the ReactorMeshParams mesh
-  freeReactorMeshParams();
+  // Store pin region ids and block names for id swap after extrusion if needed
+  // by future mesh generators
+  std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>> region_id_map{
+      {_pin_type, _region_ids}};
+  std::map<subdomain_id_type, std::vector<std::vector<std::string>>> block_name_map;
+  block_name_map[_pin_type] = _block_names;
 
-  // Update metadata at this point since values for these metadata only get set by PCCMG
-  // at generate() stage
-  if (hasMeshProperty<Real>("max_radius_meta", name() + "_2D"))
+  // Declare mesh properties that need to be moved up to the assembly level
+  if (_is_assembly)
+  {
+    declareMeshProperty(RGMB::assembly_type, _pin_type);
+    std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>> pin_region_id_map;
+    pin_region_id_map.insert(
+        std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
+            region_id_map.begin()->first, region_id_map.begin()->second));
+    declareMeshProperty(RGMB::pin_region_id_map, pin_region_id_map);
+    std::map<subdomain_id_type, std::vector<std::vector<std::string>>> pin_block_name_map;
+    pin_block_name_map.insert(std::pair<subdomain_id_type, std::vector<std::vector<std::string>>>(
+        block_name_map.begin()->first, block_name_map.begin()->second));
+    declareMeshProperty(RGMB::pin_block_name_map, pin_block_name_map);
+    declareMeshProperty(RGMB::background_block_name, std::vector<std::string>());
+    declareMeshProperty(RGMB::duct_block_names, std::vector<std::vector<std::string>>());
+  }
+  // Declare mesh properties that are only relevant to pin meshes
+  else
   {
-    const auto max_radius_meta = getMeshProperty<Real>("max_radius_meta", name() + "_2D");
-    setMeshProperty("max_radius_meta", max_radius_meta);
+    declareMeshProperty(RGMB::pin_type, _pin_type);
+    declareMeshProperty(RGMB::pin_region_ids, region_id_map);
+    declareMeshProperty(RGMB::pin_block_names, block_name_map);
```

Comment:
seems odd to keep a map that only has a single entry?

Reply:
Ive updated this now to store block names instead of a single entry map of pin_type to block names